source/server/admin/admin.cc

```diff
@@ -812,7 +811,7 @@ Http::Code AdminImpl::runCallback(absl::string_view path_and_query,
     handlerHelp(path_and_query, response_headers, response, admin_stream);
     code = Http::Code::NotFound;
   }
-
+  Memory::Utils::tryShrinkHeap();
```

Comment:
For people hitting the /stats endpoint on a regular basis this might not be a great change. I'm not sure though. At least I think there should be a release note about this?

Reply:
tryShrinkHeap() is moved from several lines above.
Previously the logic is to tryShrinkHeap() when a handler is hit. The /stats was triggering the tryShrinkHeap() before this PR.

This PR trigger the tryShrinkHeap() not only admin handler is served but also hitting http://<ip>:<adminport>/no-such-handle, and 

I also intentionally delay the shrink after /memory obtain the status *before* the shrink. Otherwise we will never know how was memory usage before the shrink.

This change is too small to put into release notes IMHO...